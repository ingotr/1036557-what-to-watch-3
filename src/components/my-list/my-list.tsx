import * as React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import SmallMovieCard from '../small-movie-card/small-movie-card';
import {getMyListMovies} from '../../reducer/data/selectors';
import {getAvatarUrl} from '../../reducer/user/selectors';
import {Operation as DataOperation} from '../../reducer/data/data';
import {MovieInterface} from '../../types';

interface Props {
  activeItem: MovieInterface | null;
  avatarUrl: string;
  onMovieFavoriteStatusClick: () => void;
  onMovieCardClick: () => void;
  onItemEnter: () => void;
  onItemLeave: () => void;
  getMyListMovies: () => void;
  movies: MovieInterface[];
}


class MyList extends React.PureComponent<Props> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getMyListMovies();
  }

  render() {
    const {movies, onMovieCardClick,
      onItemEnter, onItemLeave, avatarUrl} = this.props;
    return movies && (
      <>
        <div className="user-page">
          <header className="page-header user-page__head">
            <div className="logo">
              <Link to="/" className="logo__link">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </Link>
            </div>

            <h1 className="page-title user-page__title">My list</h1>

            <div className="user-block">
              <Link to="/mylist">
                <div className="user-block__avatar">
                  <img
                    src={avatarUrl}
                    alt="User avatar"
                    width="63"
                    height="63"
                  />
                </div>
              </Link>
            </div>
          </header>

          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            <div className="catalog__movies-list">
              {movies.map((movie) => (
                <SmallMovieCard
                  movie={movie}
                  key={movie.id}
                  onMovieCardClick={onMovieCardClick}
                  onMovieHover={onItemEnter}
                  onMovieLeave={onItemLeave}
                />
              ))}
            </div>

          </section>

          <footer className="page-footer">
            <div className="logo">
              <a className="logo__link logo__link--light">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="copyright">
              <p>Â© 2019 What to watch Ltd.</p>
            </div>
          </footer>
        </div>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  avatarUrl: getAvatarUrl(state),
  movies: getMyListMovies(state),
});

const mapDispatchToProps = (dispatch) => ({
  getMyListMovies() {
    dispatch(DataOperation.loadMyListMovies());
  }
});

export {MyList};

export default connect(mapStateToProps, mapDispatchToProps)(MyList);
