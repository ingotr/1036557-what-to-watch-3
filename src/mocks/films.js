const films = [
  {
    id: `mock-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-013`,
    title: `Macbeth`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-014`,
    title: `Aviator`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-017`,
    title: `Revenant`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-018`,
    title: `Johnny English`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
];

const movieInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Very good`,
    count: 240,
  },
  director: `Wes Andreson`,
  description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {films, movieInfo};
