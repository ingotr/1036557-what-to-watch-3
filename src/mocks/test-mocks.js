const FirstMovie = {
  TITLE: `Mock-test-The Grand Budapest Hotel`,
  GENRE: `Mock-test-Drama`,
  RELEASE_DATE: 2020,
};

const movies = [
  {
    id: `mock-test-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-013`,
    title: `Macbeth`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-014`,
    title: `Aviator`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-017`,
    title: `Revenant`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-018`,
    title: `Johnny English`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

const movieInfo = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  year: 2017,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `Mock-test-The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Mocks-Very good`,
    count: 240,
  },
  director: `Mock-Wes Andreson`,
  description: [`Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Mock-test-Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {FirstMovie, movies, movieInfo};
