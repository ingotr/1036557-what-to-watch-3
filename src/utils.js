export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const testFunc = () => {
};

export const getMovieRatingLevel = (rating) => {
  let level = ``;
  if ((rating >= 0) && (rating < 3)) {
    level = `Bad`;
  }
  if ((rating >= 3) && (rating < 5)) {
    level = `Normal`;
  }
  if ((rating >= 5) && (rating < 8)) {
    level = `Good`;
  }
  if ((rating >= 8) && (rating < 10)) {
    level = `Very good`;
  }
  if (rating === 10.0) {
    level = `Awesome`;
  }
  return level;
};
