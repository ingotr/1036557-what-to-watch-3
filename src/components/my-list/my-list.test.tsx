import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {MyList} from './my-list';
import {movies} from '../../mocks/test-mocks';
import {BrowserRouter} from 'react-router-dom';

import {testFunc} from '../../utils';

it(`Render MyList`, () => {
  const main = renderer
          .create(
              <BrowserRouter >
                <MyList
                  activeItem={null}
                  avatarUrl=""
                  movies={movies}
                  onMovieCardClick={testFunc}
                  onMovieFavoriteStatusClick={testFunc}
                  onItemEnter={testFunc}
                  onItemLeave={testFunc}
                  getMyListMovies={testFunc}
                />
              </BrowserRouter>, {
                createNodeMock: () => {
                  return {};
                }
              }
          )
    .toJSON();

  expect(main).toMatchSnapshot();
});
