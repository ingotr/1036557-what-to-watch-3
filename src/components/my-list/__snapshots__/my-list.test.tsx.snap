// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyList 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src=""
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-011"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-012"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-013"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-014"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-015"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-016"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-017"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-018"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-019"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-020"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-021"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-022"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-023"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-024"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-025"
            onClick={[Function]}
          />
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          height="175"
          width="250"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/mock-test-026"
            onClick={[Function]}
          />
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
